lib/rules/no-array-prototype-extensions.js

```diff
@@ -152,12 +152,29 @@ function variableNameToWords(name) {
  * @param {Object} fixer The ESLint fixer object which will be used to apply fixes.
  * @returns {Object|[]}
  */
-function applyFix(callExpressionNode, fixer) {
+function applyFix(callExpressionNode, fixer, context) {
   const propertyName = callExpressionNode.callee.property.name;
 
   switch (propertyName) {
     case 'any':
       return fixer.replaceText(callExpressionNode.callee.property, 'some');
+    case 'filterBy': {
+      const callArgs = callExpressionNode.arguments;
+      const sourceCode = context.getSourceCode();
+
+      return [
+        fixer.replaceText(callExpressionNode.callee.property, 'filter'),
+        fixer.insertTextBefore(sourceCode.ast, "import { get } from '@ember/object';\n"),
```

Comment:
We should only add this import statement if `get` isn't already imported. There are a bunch of other rules that have `importedGetName` for tracking the name it's imported under and you can use that to know if we need to add the import.

Reply:
@bmish Isn't there an existing lint rule which fixes duplicate import declarations?

Reply:
There are some partially-auto-fixable rules that could help.

* https://eslint.org/docs/latest/rules/no-duplicate-imports (not in ESLint's recommended config)
* https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-duplicates.md (enabled only as [warn](https://github.com/import-js/eslint-plugin-import/blob/main/config/recommended.js))

But it looks like they aren't enabled by default? And they might not always fix the issue? I would only be okay with relying on another lint rule to fix it for us if we know it will be enabled in a standard Ember app and if we know it will fix all of these duplicates we could end up with:

```js
import { get } from '@ember/object';
import { get } from '@ember/object';
```

```js
import { get } from '@ember/object';
import { set, get } from '@ember/object';
```

```js
import { get } from '@ember/object';
import EmberObject, { get } from '@ember/object';
```

Reply:
Fair enough. Used `importedGetName` to add import statement only it it's not been imported already.